registry/repository.go

```diff
@@ -84,11 +130,29 @@ func (c *repositoryClient) FetchSignatureBlob(ctx context.Context, desc ocispec.
 // linked signature envelope blob. Upon successful, PushSignature returns
 // signature envelope blob and manifest descriptors.
 func (c *repositoryClient) PushSignature(ctx context.Context, mediaType string, blob []byte, subject ocispec.Descriptor, annotations map[string]string) (blobDesc, manifestDesc ocispec.Descriptor, err error) {
-	blobDesc, err = oras.PushBytes(ctx, c.Repository.Blobs(), mediaType, blob)
+	// sanity check
+	if annotations == nil {
+		return ocispec.Descriptor{}, ocispec.Descriptor{}, errors.New("pushing signature blob, but got nil annotations map")
+	}
+	if _, ok := annotations[envelope.AnnotationX509ChainThumbprint]; !ok {
+		return ocispec.Descriptor{}, ocispec.Descriptor{}, fmt.Errorf("pushing signature blob, but annotations map missing field %q", envelope.AnnotationX509ChainThumbprint)
+	}
+	signingTime, err := envelope.SigningTime(blob, mediaType)
 	if err != nil {
 		return ocispec.Descriptor{}, ocispec.Descriptor{}, err
 	}
-
+	annotations[ocispec.AnnotationCreated] = signingTime.Format(time.RFC3339)
```

Comment:
`PushSignature` should not generate any annotations. Those annotations are generated elsewhere.